src/history/hs.c

```diff
@@ -1617,6 +1644,12 @@ __hs_delete_key_from_pos(
         WT_RET(__wt_compare(session, NULL, &hs_key, key, &cmp));
         if (cmp != 0)
             break;
+
+        /* If the stop time pair on the tombstone in the history store is already globally visible
+         * we can skip it. */
+        if (__wt_txn_visible_all(
+              session, hs_cbt->upd_value->tw.stop_txn, hs_cbt->upd_value->tw.durable_stop_ts))
```

Comment:
It does fall out neatly that it's just a visible check and a continue. It's sad that we need to repeat the pattern four times, but as we've talked about restructuring the code to avoid that is a task for another day.